### PR TITLE
feat: Added PULL REQUEST TEMPLATE 🛠

### PR DESCRIPTION
**Pull request by MrKrishnaAgarwal**
Signed-off-by: Krishna Agarwal <dmkrishna.agarwal@gmail.com>

cc @samhita-alla

### CODE REVIEW

**Comment by MrKrishnaAgarwal** [comment]
#take

**Comment by MrKrishnaAgarwal** [comment]
@samhita-alla Please look on this PR :)

**Comment by MrKrishnaAgarwal** [comment]
> How does the PR template look on the UI? Would you be able to share a screenshot?

Sure, Just a moment

**Comment by MrKrishnaAgarwal** [comment]
I have mentioned everything there in details using comments so that the user can fill and respond as much as he/she can.

A filled example of this PR Template, in this example the non-applicable sections are removed by the user. 



I hope that makes the point clear and you can go to the files changed tab and see the file yourself there :)

@samhita-alla

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+<!--
+[x] - Correct; marked as done
+[X] - Correct; marked as done
+[ ] - Not correct; marked as **not** done
+-->
+
+## Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->
+
+- [ ] My code follows the code style of this project.
+- [ ] My change requires changes to the documentation.
```
Comment (samhita-alla) [inline .github/PULL_REQUEST_TEMPLATE.md:29]: Can you remove this?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] I have updated the documentation accordingly.
+- [ ] All new and existing tests passed.
+- [ ] This PR does not contain plagiarized content.
+- [ ] The title of my pull request is a short description of the requested changes.
```
Comment (samhita-alla) [inline .github/PULL_REQUEST_TEMPLATE.md:33]: Can you remove these checks?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] All commits are Signed-off.
```
Comment (samhita-alla) [inline .github/PULL_REQUEST_TEMPLATE.md:34]:
```suggestion
- [ ] All commits are signed-off.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+[x] - Correct; marked as done
+[X] - Correct; marked as done
+[ ] - Not correct; marked as **not** done
+-->
+
+## Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->
+
+- [ ] My code follows the code style of this project.
+- [ ] My change requires changes to the documentation.
+- [ ] I have updated the documentation accordingly.
```
Comment (samhita-alla) [inline .github/PULL_REQUEST_TEMPLATE.md:30]:
```suggestion
- [ ] I updated the documentation accordingly.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] All new and existing tests passed.
+- [ ] This PR does not contain plagiarized content.
+- [ ] The title of my pull request is a short description of the requested changes.
+- [ ] All commits are Signed-off.
+
+## Screenshots
+
+<!-- Add all the screenshots which support your changes -->
+
+## Note to Reviewers
```
Comment (samhita-alla) [inline .github/PULL_REQUEST_TEMPLATE.md:40]:
```suggestion
## Note to reviewers
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+<!-- List all the proposed changes in your PR -->
+
+<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->
+<!--
+[x] - Correct; marked as done
+[X] - Correct; marked as done
+[ ] - Not correct; marked as **not** done
+-->
+
+## Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->
```
Comment (samhita-alla) [inline .github/PULL_REQUEST_TEMPLATE.md:26]:
```suggestion
## Check all the applicable boxes <!-- Follow the above conventions to check the box -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+## Fixes Issue
+
+<!-- If your PR fixes an open issue, use `Closes #999` to link your PR with the issue. #999 stands for the issue number you are fixing -->
+
+<!-- Remove this section if not applicable -->
+
+<!-- Example: Closes #31 -->
+
+## Changes proposed (Describe Changes)
```
Comment (samhita-alla) [inline .github/PULL_REQUEST_TEMPLATE.md:15]:
```suggestion
## Describe your changes
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+<!--
+Thank you for sending the PR! 
+Please fill the applicable details below
+Happy contributing!
+-->
+
+## Fixes Issue
```
Comment (samhita-alla) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
## Tracking issue
```

**Comment by MrKrishnaAgarwal** [comment]
@samhita-alla,
All requested changes have been made.
Kindly check!

**Review by samhita-alla — APPROVED** [review]

**Pull request merged**